### PR TITLE
KEYCLOAK-9820

### PR DESCRIPTION
**Pull request by abstractj**
@hmlnarik @stianst after your approval on this, I'm going to update the prod repo. Here some considerations:

* There were two dependencies which required some upgrades Semistandard and Istanbul. `Semistandard` is used to scan for bad practices into the code and also code formatting, while `Istanbul` is used for code coverage.
* After the upgrade `Semistandard` reported several bad smells into our codebase, including formatting and error handling. Both were fixed. That's the reason why this PR touches 14 files.
* Istanbul was deprecated and moved to `nyc` project
* For semistandard there are still some issues reported with npm audit. To make sure that the versions are locked and we're using the most up to date dependencies a new `package-lock.json` file was generated.
* You can see that there are no issues anymore by running `npm audit` 

If there are any questions, please let me know.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -2274,7 +2274,7 @@
       "dependencies": {
         "es6-promise": {
           "version": "3.0.2",
-          "resolved": "https://registry.npmjs.org/es6-promise/-/es6-promise-3.0.2.tgz",
+          "resolved": "http://registry.npmjs.org/es6-promise/-/es6-promise-3.0.2.tgz",
```
Comment (hmlnarik) [inline package-lock.json]: Why is there the change from https to http? (same with readable-stream dep)

Comment (abstractj) [reply package-lock.json]: @hmlnarik because that's the standard in new versions of npm. Do you see any blockers?

Comment (stianst) [reply package-lock.json]: Strange. So while the rest of the world is going to https always NPM is for some reason going backwards?

Comment (hmlnarik) [reply package-lock.json]: No blockers, the switch from https to unsecured http just seemed a typo to me (particularly with almost of the other dependencies still using https)

Comment (abstractj) [reply package-lock.json]: @hmlnarik @stianst let me rephrase myself. Not a standard, but known bug https://github.com/npm/npm/issues/20719

**Comment by abstractj** [comment]
@hmlnarik it's not bad, it's awful. Until npm fix it, we can leave it as is.
